pkg/config/expand.go

```diff
@@ -164,6 +165,7 @@ func getModuleInputMap(inputs []modulereader.VarInfo) map[string]string {
 func useModule(
 	mod *Module,
 	useMod Module,
+	useModGroupID string,
 	modInputs map[string]string,
 	useOutputs []modulereader.VarInfo,
 	settingsInBlueprint []string,
```

Comment:
No we cannot because `useModule` is being called iteratively over every element of the `use` list in `applyUseModules` below. This will (for better/worse) iteratively modify `mod.Settings` in-place. One of the mechanisms by which it modifies `mod.Settings` is to build up a list  when the corresponding `Input` is of type `list`.

A more accurate name for this variable may be `settingsToIgnore` with the clarification about expectation that it would be used to ignore explicit settings in the blueprint.

Reply:
nit: I think here and in other comments you have some constructive ideas on how to change the signature of this function. I know it is not directly needed for your work, but I would strongly ask/recommend that in a follow on pr you refactor the signature of this function with your suggestions.

Reply:
In this PR, I would strongly suggest that you look at cyclomatic complexity as a proxy for code quality and then keep it <= 7. That should force you to rethink this code. For instance, I feel this function is now way too complex.

Reply:
The cyclomatic complexity of this function is 7 before and after the PR.  During PR review, we were able to make a change that dedented by the block by "exiting early" (a 1-for-1 change in cyclomatic complexity but hopefully more conceptually clear and readable).

That said, while developing these changes, I gave consideration to altering the block that now reads... `if isInputList { ... else if !setByUse ... }`. In broad strokes, I considered:

- a refactor that would put `if setByUse` as the outer block
- improving commenting and variable naming for clarity

I concluded that improving the variable names and commenting was the best alternative since it would be least likely to introduce unintentional subtle behavioral changes (i.e. a mistake). The ultimate complexity here stems from the fact that our `Settings` are just `interface{}` and we need to initialize settings whose input types are lists to ` []interface{}{}` and then append to them rather than overwrite them.

I can see value in a `func` that initializes a setting of type list. I'm not sure it will reduce the cyclomatic complexity but it is something that ought to have one correct implementation.

Reply:
I think the long comment might be a better fit as a comment to the func itself rather than to the block.

LMK.

Reply:
If you are proposing moving all the comments into a single block at the top, I don't know if that is better. My advice would be to merge as is and then spend 4 hours to make this section incrementally better in one or more separate prs. My suggestion would be:
- Break out a few functions to make the code self documenting
- Change the function signature with the ideas you have proposed
- Add some minimal function descriptions that briefly describe the purpose of the functions

Reply:
I agree with these changes in broad strokes and will submit some/all of them as a separate PR.